### PR TITLE
Wan replication config for ICache

### PR DESCRIPTION
**Pull request by emrahkocaman**
Prepares CacheConfig for WAN replication

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/config/WanReplicationRef.java [hunk 1]
```diff
@@ -15,10 +15,18 @@
...
+
+import java.io.IOException;
+import java.io.Serializable;
+
 /**
  * Configuration for Wan target replication reference
  */
-public class WanReplicationRef {
+public class WanReplicationRef implements DataSerializable, Serializable {
     private String name;
```
Comment (mesutcelik) [inline hazelcast/src/main/java/com/hazelcast/config/WanReplicationRef.java]: why does this both implements DataSerializable and Serializable?

Comment (emrahkocaman) [inline hazelcast/src/main/java/com/hazelcast/config/WanReplicationRef.java]: @mesutcelik , `CacheConfiguration` interface extends `CompleteConfiguration` interface (from JCache API) which is `Serializable`. So all of its non-transient fields needs to be `Serializable`. (like `CacheEvictionConfig` and `NearCacheConfig`)

**Comment by mesutcelik** [comment]
failure....

is this related?

```
ReadWriteThroughJCacheTests.test_loadall_readthrough
```

**Comment by mesutcelik** [comment]
verify

**Comment by emrahkocaman** [comment]
verify

**Comment by mesutcelik** [comment]
@serkan-ozal , 
Can you please have a look?

**Comment by serkan-ozal** [comment]
@mesutcelik @emrahkocaman sure, let me look

**Comment by serkan-ozal** [comment]
@emrahkocaman ```WanReplicationRef``` must also be handled in ```com.hazelcast.config.CacheConfigReadOnly```

**Comment by emrahkocaman** [comment]
Thanks @serkan-ozal , I've changed `CacheConfigReadOnly` to handle `WanReplicationRef`. Can you have a look at this again?

**Comment by serkan-ozal** [comment]
:+1:

**Pull request merged**